### DISCUSSION TITLE
CONNECT packet must have at least 7 bytes.

### DISCUSSION THREAD

**Discussion by zzhanxianzhi** [dotnet/MQTTnet#1808] [Q&A]
After connecting to the client, it frequently disconnects and prompts Error while processing CONNECT packet, CONNECT packet must have at least 7 bytes. My code
                MqttServerOptionsBuilder mqttServerOptionsBuilder = new MqttServerOptionsBuilder();  
                mqttServerOptionsBuilder.WithDefaultEndpoint();
                mqttServerOptionsBuilder.WithDefaultEndpointBoundIPAddress(IPAddress.Parse(MqttService.ip));  
                mqttServerOptionsBuilder.WithDefaultEndpointPort(MqttService.port);                           
                //mqttServerOptionsBuilder.WithEncryptedEndpointPort(port);                       
                mqttServerOptionsBuilder.WithPersistentSessions(MqttService.withPersistentSessions);  
                mqttServerOptionsBuilder.WithDefaultCommunicationTimeout(TimeSpan.FromMilliseconds(60000));

                mqttServerOptionsBuilder.WithKeepAlive();
                mqttServerOptionsBuilder.WithTcpKeepAliveTime(7200);
                mqttServerOptionsBuilder.WithTcpKeepAliveRetryCount(5);
                mqttServerOptionsBuilder.WithTcpKeepAliveInterval(60);

                mqttServerOptionsBuilder.WithConnectionBacklog(MqttService.MaxNum);                    
                MqttServerOptions mqttServerOptions = mqttServerOptionsBuilder.Build();
                var mqttFactory = new MqttFactory(new ConsoleLogger(channle));                     
                MqttService._MqttServer = mqttFactory.CreateMqttServer(mqttServerOptions);

**Comment by gfoidl** [comment]
For reference: https://github.com/dotnet/MQTTnet/issues/1809 (same issue).
